src/qubex/experiment/mixin/characterization_mixin.py

```
@@ -1814,12 +1912,12 @@ def estimate_control_amplitude(
     def qubit_spectroscopy(
         self,
         target: str,
-        frequency_range: ArrayLike = np.arange(6.5, 9.5, 0.002),
+        frequency_range: ArrayLike | None = None,
         power_range: ArrayLike = np.arange(-60, 5, 5),
```

Ruff (0.8.2)

Do not perform function call `np.arange` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)